Random seed of the ExecutableModel can now be set through API functions (With rebasing the develop branch)

adelhpour opened the pull request:
- SEED is added as a SelectionType to the list of SelectionRecords
    
    - 'seed' is recognized as the id for SEED in the SelectionRecords
    
    - the getValue function of API now returns the value of getRandomSeed function of the model if it is asked to return the 'seed' value
    
    - the setValue function of API now sets the value of getRandomSeed function of the model if it is asked to set the 'seed' value

luciansmith committed:
Update version number to 2.6.0

Have to do this because of piecewise change--the save state is now different.

luciansmith committed:
Clear off space as we build.

It *might* be that the rrplugins distributions were larger than they needed to be because we didn't clear out the build/ directory before building the rrplugins wheels.  But more importantly, the ubuntu builds were dying because the disk was filling up, so deleting the build directory should help.

Also updated numpy requirement to 1.26, since that's what the build system is using.

luciansmith committed:
Actually remove build directory.

adelhpour committed:
- SEED is added as a SelectionType to the list of SelectionRecords

- 'seed' is recognized as the id for SEED in the SelectionRecords

- the getValue function of API now returns the value of getRandomSeed function of the model if it is asked to return the 'seed' value

- the setValue function of API now sets the value of getRandomSeed function of the model if it is asked to set the 'seed' value

adelhpour committed:
- the bug with getting the default seed from Config::RANDOM_SEED vale is fixed

- get 'seed' value now returns the Config::RANDOM_SEED value

- set 'seed' value now sets the Config::RANDOM_SEED value

- after setting 'seed' value regenerateModel is called

adelhpour committed:
- setSeed and getSeed function are added to the API function.

- SEED is removed from the selections list, so it is no longer possible to set and get its value using setValue and getValue

- the overridden setValue function in the Gillespie integrator is now deprecated since there is no need to set its seed value independently

- setSeed and getSeed function in hte rrc_api.cpp are now deprecated

- the bug with getting RANDOM_SEED value in Random class is fixed again

adelhpour committed:
In both OUTPUT_FILE_VARIABLE_TIMESTEP and OUTPUT_FILE_FIXED_TIMESTEP tests, new method of setting seed value is used

adelhpour committed:
In both OUTPUT_FILE_VARIABLE_TIMESTEP and OUTPUT_FILE_FIXED_TIMESTEP tests, new method of setting seed value is used

adelhpour committed:
Merge remote-tracking branch 'origin/setting-random-seed-for-executable-model-with-develop-rebase' into setting-random-seed-for-executable-model-with-develop-rebase

# Conflicts:
#	test/c_api_rrtests/OtherRRTestFileTests.cpp

adelhpour committed:
Merge remote-tracking branch 'origin/setting-random-seed-for-executable-model-with-develop-rebase' into setting-random-seed-for-executable-model-with-develop-rebase

# Conflicts:
#	test/c_api_rrtests/OtherRRTestFileTests.cpp

adelhpour committed:
Merge remote-tracking branch 'origin/setting-random-seed-for-executable-model-with-develop-rebase' into setting-random-seed-for-executable-model-with-develop-rebase

adelhpour committed:
brining back setValue function for GillespieIntegrator

adelhpour committed:
brining back changed tests to their previous situation

adelhpour committed:
function setSeed and getSeed definitions are removed from rrc_api.h

adelhpour committed:
RESET_RANDOM_SEED is added to the list of Config Settings

adelhpour committed:
- setSeed function now receives a second argument that sets the value of Config::RESET_RANDOM_SEED

- reset function is updated so that it resets the value of seed if Config::RESET_RANDOM_SEED is set to true

- an API function called resetSeed that resets the Config::RANDOM_SEED value to -1 and Config::RESET_RANDOM_SEED value to false is added

adelhpour committed:
rrc_api.cpp function called 'resetAll' is updated so that it also calls resetSeed RoadRunner API function

adelhpour committed:
in setSeed function, reset function is called after regenerate function to use the flags to reset all the values while setSeed function is called

adelhpour committed:
SetSeed_Get_Seed_Value test is added to the list of ModelAnalysisTests to test the newly added 'setSeed' and 'getSeed' functions

adelhpour committed:
- a flag, 'resetModel', that determines if setSeed function needs to reset the model or not is added to 'setSeed' arguments

- a test is written to check if the flag works properly

adelhpour committed:
- a flag, 'resetModel', that determines if setSeed function needs to reset the model or not is added to 'setSeed' arguments

- a test is written to check if the flag works properly

adelhpour committed:
test file uniform_distrib_model.xml is added to the ModelAnalysis tests

adelhpour committed:
test file uniform_distrib_model.xml is added to the ModelAnalysis tests

adelhpour committed:
removing the not equal part in the SetSeed_Get_Seed_Value_With_Reset_Model test as there is no must they be different all the time

adelhpour committed:
removing the not equal part in the SetSeed_Get_Seed_Value_With_Reset_Model test as there is no must they be different all the time

adelhpour committed:
- 'resetSeed' flag is removed from the 'setSeed' function

- RESET_RANDOM_SEED is removed from the list of Config Settings

adelhpour committed:
- 'resetSeed' flag is removed from the 'setSeed' function

- RESET_RANDOM_SEED is removed from the list of Config Settings

adelhpour committed:
Merge remote-tracking branch 'origin/develop' into develop

adelhpour committed:
Merge remote-tracking branch 'origin/setting-random-seed-for-executable-model-with-develop-rebase' into setting-random-seed-for-executable-model-with-develop-rebase

adelhpour committed:
add the docstring for 'setSeed' and 'getSeed' python functions to throw a warning about integrator option being deprecated

luciansmith commented on source/rrRoadRunner.cpp:
```diff
@@ -7186,6 +7184,32 @@ namespace rr {
+        }
+        else {
+            impl->model->setRandomSeed(seed);
+            if (getIntegrator()->getName() == "gillespie")
```
Instead of checking the current integrator, I think we need to find the gillepie integrator (if it exists) and set the seed.

luciansmith replied:
i.e. getIntegrator("gillespie"), and then call setSeed on it if it exists.

adelhpour committed:
setEngineSeed of GillespieIntegrator now checks if the value is equal to -1, it sets a random value for the engine seed.

adelhpour committed:
from now on, gillespie integrator considers the -1 value as the random seed value

adelhpour committed:
based on @lucianpsmith comment, setSeed function now get gillespie integrator and sets its seed value.

luciansmith committed:
Update python docs; always set gillespie integrator seed.

Don't call 'getIntegratorByName', since that creates a new one if it didn't exist yet.  Just set one if it already exists.

luciansmith committed:
Need to include climits in Ubuntu.

adelhpour committed:
- it is now possible to get the actual used seed using the 'getSeed' function

- Some Gillespie tests ard disabled

adelhpour committed:
change the method to update the gillespie integrator seed

adelhpour committed:
fix the bug to get unsigned int from a settings value

adelhpour committed:
revert back 'defaultSeed' function.

adelhpour committed:
'seedValue' function is introduced to convert seedSetting into a std::uint64_t value

adelhpour committed:
disable Gillespie_Random_Seed test.

adelhpour committed:
commented asserts in model analysis tests are reverted

adelhpour committed:
set the seed for stochastic simulator in stochastic test suite

adelhpour committed:
set maximum number of allowed wrong number of tries to 10 to see how many of stochastic pass

adelhpour committed:
assert statement is removed from countWrongMeans function as it already checks at leas one time if nmean_wrong is less than 3.

luciansmith committed:
Remove tab characters and fix formatting.

(used the autoformat bit of VS)

adelhpour committed:
Merge remote-tracking branch 'origin/setting-random-seed-for-executable-model-with-develop-rebase' into setting-random-seed-for-executable-model-with-develop-rebase

luciansmith committed:
Tell the user in INFORMATION what seed was actually used.

luciansmith committed:
Seed updates.

* Refuse to run gillespie simulator on models with rate rules.
* Warn when running gillespie simulator on models with 'time' in event triggers, or with event delays.
* Remove some gillespie tests that checked 'time' and delays in events; they'll never work.
* getSeed now always returns int64.
* Hopefully fix remaining tests.

luciansmith committed:
Couple small edits.
